src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/SdkDirectoryWorkloadManifestProvider.cs

```diff
@@ -169,28 +169,24 @@ bool TryGetWorkloadSet(string workloadSetVersion, out WorkloadSet? workloadSet)
                 }
             }
 
+            _installStateFilePath = Path.Combine(WorkloadInstallType.GetInstallStateFolder(_sdkVersionBand, _sdkRootPath), "default.json");
+            var installState = InstallStateContents.FromPath(_installStateFilePath);
             if (_workloadSet is null)
             {
-                var installStateFilePath = Path.Combine(WorkloadInstallType.GetInstallStateFolder(_sdkVersionBand, _sdkRootPath), "default.json");
-                if (File.Exists(installStateFilePath))
+                if (!string.IsNullOrEmpty(installState.WorkloadVersion))
                 {
-                    var installState = InstallStateContents.FromPath(installStateFilePath);
-                    if (!string.IsNullOrEmpty(installState.WorkloadVersion))
+                    if (!availableWorkloadSets.TryGetValue(installState.WorkloadVersion!, out _workloadSet))
                     {
-                        if (!TryGetWorkloadSet(installState.WorkloadVersion!, out _workloadSet))
-                        {
-                            throw new FileNotFoundException(string.Format(Strings.WorkloadVersionFromInstallStateNotFound, installState.WorkloadVersion, installStateFilePath));
-                        }
+                        throw new FileNotFoundException(string.Format(Strings.WorkloadVersionFromInstallStateNotFound, installState.WorkloadVersion, _installStateFilePath));
                     }
-
-                    //  Note: It is possible here to have both a workload set and loose manifests listed in the install state.  This might happen if there is a
-                    //  third-party workload manifest installed that's not part of the workload set
-                    _manifestsFromInstallState = installState.Manifests is null ? null : WorkloadSet.FromDictionaryForJson(installState.Manifests, _sdkVersionBand);
-                    _installStateFilePath = installStateFilePath;
                 }
+
+                //  Note: It is possible here to have both a workload set and loose manifests listed in the install state.  This might happen if there is a
+                //  third-party workload manifest installed that's not part of the workload set
+                _manifestsFromInstallState = installState.Manifests is null ? null : WorkloadSet.FromDictionaryForJson(installState.Manifests!, _sdkVersionBand);
             }
 
-            if (_workloadSet == null && availableWorkloadSets.Any())
+            if (_workloadSet == null && installState.UseWorkloadSets == true && availableWorkloadSets.Any())
```

Comment:
`UseWorkloadSets` is a nullable bool.  I think if it's null (IE not explicitly set), then we probably want the existing behavior where it will use workload sets.  That might also make the test changes unnecessary.

```suggestion
            if (_workloadSet == null && installState.UseWorkloadSets != false && availableWorkloadSets.Any())
```

Reply:
Yes, I was also thinking it probably wouldn't affect the behavior in most cases.

I think the main argument for changing it is so that we don't have to remember to go back and change the code here when we change the default to use workload sets.  Another way to achieve that could be to add an `InstallStateContents.UseWorkloadSetsDefaultValue` constant, and wherever we read the `UseWorkloadSets` property, do something like `installState.UseWorkloadSets ?? InstallStateContents.UseWorkloadSetsDefaultValue`.

Reply:
I support having a default value like that, but it's a bit late for 8.0.4xx, and hopefully we won't change the default after we change it in 9.0.1xx